src/state/nftMarket/helpers.ts

```diff
@@ -27,3 +27,27 @@ export const getCollections = async (): Promise<any[]> => {
     return []
   }
 }
+
+export const getNftsFromCollection = async (collectionAddress: string): Promise<any[]> => {
```

Comment:
The query will be faster if you fetch the `Collection` entity and fetch all subsequent tokens, such as:

```graphql
{
  collection(id: "${collectionAddress}") {
    id
    NFTs {
      currentSeller
      isTradable
      tokenId
      metadataUrl
    }
  }
}
```

Also, you need to make sure the `collectionAddress` is lowercased

Reply:
Please, also leverage the filter parameters included within graphql library such as for the Prediction: https://github.com/pancakeswap/pancake-frontend/blob/develop/src/state/predictions/helpers.ts#L257